packages/notifications/push/amplify_push_notifications/lib/src/amplify_push_notifications_impl.dart

```diff
@@ -49,15 +68,27 @@ class AmplifyPushNotifications extends PushNotificationsPluginInterface {
   AmplifyPushNotifications({
     required ServiceProviderClient serviceProviderClient,
     required Future<void> Function() backgroundProcessor,
```

Comment:
Why is this required?

Reply:
This class should probably be `abstract`. I don't see when it would be instantiated directly ever.

Reply:
Can you update to make this abstract?